ArrayToFirstClassCallableRector: "Scope not available on Array_" when array arg precedes named arg in variadic call

# Bug Report

| Subject        | Details |
| :------------- | :------ |
| Rector version | 2.3.9   |

`ArrayToFirstClassCallableRector` crashes with "Scope not available on Array_ node" when an array literal argument precedes a named argument in a call to a variadic function inside a class method.

## Minimal PHP Code Causing Issue

Reproduced on https://getrector.com/demo with `ArrayToFirstClassCallableRector` selected.
https://getrector.com/demo/cbeb983a-fbe4-4e7b-a7af-4f5787cffe88

```php
<?php

declare(strict_types=1);

function foo(mixed ...$args): void {}

class ReproCase
{
    public function test(): void
    {
        foo('first', ['key' => 'value'], c: 'test');
    }
}
```

Debug output (`vendor/bin/rector process --debug`):
```
[file] rector-repro.php

 [ERROR] Could not process "rector-repro.php" file, due to:
         "System error: "Scope not available on "PhpParser\Node\Expr\Array_"
         node. Fix scope refresh on changed nodes first"

         Stack trace:
         #0 vendor/rector/rector/rules/Php81/Rector/Array_/ArrayToFirstClassCallableRector.php(98):
         Rector\PHPStan\ScopeFetcher::fetch(Object(PhpParser\Node\Expr\Array_))
         #1 vendor/rector/rector/src/Rector/AbstractRector.php(106):
         Rector\Php81\Rector\Array_\ArrayToFirstClassCallableRector->refactor(Object(PhpParser\Node\Expr\Array_))
         #2 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(158):
         Rector\Rector\AbstractRector->enterNode(Object(PhpParser\Node\Expr\Array_))
         #3 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(242):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverseNode(Object(PhpParser\Node\Arg))
         #4 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(146):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverseArray(Array)
         #5 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(186):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverseNode(Object(PhpParser\Node\Expr\FuncCall))
         #6 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(242):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\Expression))
         #7 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(146):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverseArray(Array)
         #8 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(242):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\ClassMethod))
         #9 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(146):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverseArray(Array)
         #10 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(242):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\Class_))
         #11 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(146):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverseArray(Array)
         #12 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(242):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverseNode(Object(Rector\PhpParser\Node\FileNode))
         #13 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(90):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverseArray(Array)
         #14 vendor/rector/rector/src/Application/FileProcessor.php(96):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverse(Array)
```

All five conditions are required to trigger the crash. Removing any one makes it pass:

1. Function has a **variadic** parameter (`mixed ...$args`)
2. The call is inside a **class method**
3. There is a **positional argument before** the array
4. An **array literal** is passed as an argument
5. A **named argument** follows the array

| Variant | Result |
|---------|--------|
| `foo('first', ['key' => 'value'], c: 'test')` | **Crash** |
| `foo('first', ['key' => 'value'], 'test')` | OK (no named arg) |
| `foo(['key' => 'value'], c: 'test')` | OK (no positional arg before array) |
| `foo('first', ['key' => 'value'], c: 'test')` with `function foo(mixed $a, array $b, mixed $c)` | OK (non-variadic) |
| Same call outside a class (top-level) | OK |

## Expected Behaviour

Rector should skip the `['key' => 'value']` array (it's not a callable) and finish without errors.

--- a/rules-tests/Php81/Rector/Array_/ArrayToFirstClassCallableRector/Fixture/skip_normal_array_with_variadic.php.inc
+++ b/rules-tests/Php81/Rector/Array_/ArrayToFirstClassCallableRector/Fixture/skip_normal_array_with_variadic.php.inc
@@ -1,0 +1,13 @@
+<?php
+
+namespace Rector\Tests\Php81\Rector\Array_\ArrayToFirstClassCallableRector\Fixture;
+
+final class SkipNormalArrayWithVariadic
+{
+    private function foo(mixed ...$args): void {}
+
+    public function test(): void
+    {
+        $this->foo('first', ['key' => 'value'], c: 'test');
+    }
+}

--- a/src/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php
+++ b/src/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php
@@ -492,6 +492,14 @@ final readonly class PHPStanNodeScopeResolver
         } elseif ($callLike instanceof New_ && ! $callLike->class instanceof Class_) {
             $callLike->class->setAttribute(AttributeKey::SCOPE, $mutatingScope);
         }
+
+        if ($callLike->isFirstClassCallable()) {
+            return;
+        }
+
+        foreach ($callLike->getArgs() as $arg) {
+            $arg->value->setAttribute(AttributeKey::SCOPE, $mutatingScope);
+        }
     }
 
     private function processAssign(Assign|AssignOp|AssignRef $assign, MutatingScope $mutatingScope): void

--- a/src/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php
+++ b/src/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php
@@ -6,7 +6,6 @@ namespace Rector\NodeTypeResolver\PHPStan\Scope;
 
 use Error;
 use PhpParser\Node;
-use PhpParser\Node\Arg;
 use PhpParser\Node\ArrayItem;
 use PhpParser\Node\Expr;
 use PhpParser\Node\Expr\Array_;
@@ -274,11 +273,6 @@ final readonly class PHPStanNodeScopeResolver
 
             if ($node instanceof BinaryOp) {
                 $this->processBinaryOp($node, $mutatingScope);
-                return;
-            }
-
-            if ($node instanceof Arg) {
-                $node->value->setAttribute(AttributeKey::SCOPE, $mutatingScope);
                 return;
             }
 

